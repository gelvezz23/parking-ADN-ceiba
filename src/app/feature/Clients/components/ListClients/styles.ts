import styled from 'styled-components';

export const Table = styled.table`
  width: 100%;
  height: 350px;
  text-align: center;

  padding: 20px;
`;

export const ContainerScrolling = styled.div`
  overflow: scroll;
  overflow-x: hidden;
  width: 100%;
  height: 350px;
`;
