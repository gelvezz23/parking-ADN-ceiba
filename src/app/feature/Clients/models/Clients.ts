export interface Clients {
  id: number;
  day: string;
  date: string;
  responsable: string;
  idResponsable: string;
  licensePlate: string;
}
