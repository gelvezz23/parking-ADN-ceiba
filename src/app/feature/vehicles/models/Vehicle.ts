export interface Vehicle {
  id: number;
  day: string;
  date: string;
  slot: string;
  isActive: boolean;
  responsable: string;
  type: string;
  idResponsable: string;
  licensePlate: string;
}
