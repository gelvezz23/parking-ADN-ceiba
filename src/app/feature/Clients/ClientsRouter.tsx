import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from '../../shared/components/LazyFallback';

const ClientsPage = React.lazy(() => import('./pages/Main'));

export const ClientsRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    {/* Layout compartido entre las rutas va aquÃ­ */}
    <Switch>
      <Route path="/" component={ClientsPage}></Route>
    </Switch>
  </React.Suspense>
);
