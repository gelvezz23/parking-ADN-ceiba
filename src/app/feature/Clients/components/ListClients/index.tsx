import { Clients } from './../../models/Clients';
import { ContainerScrolling } from './styles';

import { DivContainer } from 'app/shared/components/Layout/styles';
import { Link } from './../../../../shared/components/Link';
import PropTypes from 'prop-types';
import React from 'react';
import { Table } from './styles';

interface ListClientsProps {
  clients: Array<Clients>;
}

const ListClients: React.FC<ListClientsProps> = ({ clients }) => {
  return (
    <DivContainer>
      <h5>Historial de Clientes</h5>
      <ContainerScrolling>
        <Table>
          <thead>
            <tr>
              <th>
                <h2>#</h2>
              </th>
              <th>
                <h2>Nombre</h2>
              </th>
              <th>
                <h2>Identificacion</h2>
              </th>
              <th>
                <h2>Placa de vehiculo</h2>
              </th>
              <th>
                <h2>Dia de registro</h2>
              </th>
              <th>
                <h2>Hora de registro</h2>
              </th>
            </tr>
          </thead>
          <tbody>
            {clients.map((client, index) => {
              return (
                <tr key={index}>
                  <td>
                    <h5>{client.id}</h5>
                  </td>
                  <td>
                    <h5>{client.responsable}</h5>
                  </td>
                  <td>
                    <h5>{client.idResponsable}</h5>
                  </td>
                  <td>
                    <h5>{client.licensePlate}</h5>
                  </td>
                  <td>
                    <h5>{client.day}</h5>
                  </td>
                  <td>
                    <h5>{`${new Date(client.date).getHours()}:${new Date(
                      client.date
                    ).getMinutes()}:${new Date(client.date).getSeconds()}`}</h5>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </Table>
      </ContainerScrolling>
      <Link to="/">
        <h5>Volver</h5>
      </Link>
    </DivContainer>
  );
};
ListClients.propTypes = {
  clients: PropTypes.array.isRequired,
};

export default ListClients;
