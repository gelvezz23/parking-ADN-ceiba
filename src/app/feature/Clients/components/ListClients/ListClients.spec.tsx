import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { Clients } from '../../models/Clients';
import ListClients from './index';
import { BrowserRouter as Router } from 'react-router-dom';

describe('test List Clients', () => {
  let componentWrapper: RenderResult;

  const clients = [
    {
      id: 1,
      day: 'Domingo',
      date: '2021-03-29T03:17:22.364Z',
      responsable: 'Carlos',
      idResponsable: '1090503315',
      licensePlate: 'XZY88',
    },
    {
      id: 2,
      day: 'Domingo',
      date: '2021-03-29T03:22:17.908Z',
      responsable: 'Carlos',
      idResponsable: '1090503315',
      licensePlate: 'XZY888',
    },
  ];
  let componentProps: React.ComponentProps<typeof ListClients> & {
    clients: Array<Clients>;
  } = { clients };
  beforeEach(() => {
    componentProps = {
      clients,
    };
    componentWrapper = render(
      <Router>
        <ListClients clients={clients} />
      </Router>
    );
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should render 3 td', () => {
    const elem = componentWrapper.container;
    const tr = elem.querySelectorAll('tr');
    const expected = 3;
    expect(tr.length).toBe(expected);
  });
});
