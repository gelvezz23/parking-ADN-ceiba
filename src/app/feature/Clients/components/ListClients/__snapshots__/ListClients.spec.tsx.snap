// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test List Clients should match snapshot 1`] = `
<div>
  <div
    class="sc-AxiKw gbEGRu"
  >
    <h5>
      Historial de Clientes
    </h5>
    <div
      class="sc-AxirZ iYlrcd"
    >
      <table
        class="sc-AxjAm dKjXjL"
      >
        <thead>
          <tr>
            <th>
              <h2>
                #
              </h2>
            </th>
            <th>
              <h2>
                Nombre
              </h2>
            </th>
            <th>
              <h2>
                Identificacion
              </h2>
            </th>
            <th>
              <h2>
                Placa de vehiculo
              </h2>
            </th>
            <th>
              <h2>
                Dia de registro
              </h2>
            </th>
            <th>
              <h2>
                Hora de registro
              </h2>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <h5>
                1
              </h5>
            </td>
            <td>
              <h5>
                Carlos
              </h5>
            </td>
            <td>
              <h5>
                1090503315
              </h5>
            </td>
            <td>
              <h5>
                XZY88
              </h5>
            </td>
            <td>
              <h5>
                Domingo
              </h5>
            </td>
            <td>
              <h5>
                22:17:22
              </h5>
            </td>
          </tr>
          <tr>
            <td>
              <h5>
                2
              </h5>
            </td>
            <td>
              <h5>
                Carlos
              </h5>
            </td>
            <td>
              <h5>
                1090503315
              </h5>
            </td>
            <td>
              <h5>
                XZY888
              </h5>
            </td>
            <td>
              <h5>
                Domingo
              </h5>
            </td>
            <td>
              <h5>
                22:22:17
              </h5>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <a
      class="sc-AxhCb lbSIQY"
      href="/"
    >
      <h5>
        Volver
      </h5>
    </a>
  </div>
</div>
`;
